libr/core/cmd_write.c

```diff
@@ -1239,7 +1304,8 @@ static int cmd_write(void *data, const char *input) {
 			break;
 		case ' ':
 			if (size>0) {
-				r_io_use_desc (core->io, core->file->desc);
+				if (core->file->desc)
+					r_io_desc_use (core->io, core->file->desc->fd);
```

Comment:
what about having r_io_desc_use and r_io_desc_use_fd ? and revert all those changes in the API?